### DISCUSSION TITLE
Are all methods in Toga.MainWindow implemented on iOS?

### DISCUSSION THREAD

**Discussion by Kaiyiccc** [beeware/toga#1320] [Q&A]
Tried to build [this example: dialogs.py](https://github.com/beeware/toga/tree/master/examples/dialogs) as an app on iOS to test the dialogs.

Seems like the following methods don't work on iOS:

> main_window.question_dialog()
> main_window.open_file_dialog()
> main_window.select_folder_dialog()
> main_window.save_file_dialog()

Does that mean the MainWindow methods interacting with iOS filesystem are not implemented?

We're trying to read in a file from the "Files" App on iOS. Would there be some work arounds?

Thanks a lot.

**Comment by Kaiyiccc** [comment]
Found the source [here](https://github.com/beeware/toga/blob/5ba9def18a73981882d30ed0fd5a0ac3258348b6/src/iOS/toga_iOS/window.py) and [here](https://github.com/beeware/toga/blob/5ba9def18a73981882d30ed0fd5a0ac3258348b6/src/iOS/toga_iOS/dialogs.py#L133), seems like they are not implemented.

**Reply by freakboy3742** [reply]
Correct - file dialogs aren't currently implemented on iOS. 

File handling on mobile platforms is different to desktop, as you can't "just write to a file"; the interaction is almost always with a cloud service (like the Files app, Dropbox, etc). 

Normal Python file IO operations will work, but with some restrictions. There are limits on the locations where you can open files, and the data is only visible "in app" - there's no cross-app sharing.

Eventually, we would like to offer file APIs that integrate with Files (and other cloud stores), providing a file API that is as seamless as possible; however, that's likely a way off.
